sympy/physics/vector/point.py

```diff
@@ -483,19 +483,46 @@ def vel(self, frame):
         Examples
         ========
 
-        >>> from sympy.physics.vector import Point, ReferenceFrame
+        >>> from sympy.physics.vector import Point, ReferenceFrame, dynamicsymbols
         >>> N = ReferenceFrame('N')
         >>> p1 = Point('p1')
         >>> p1.set_vel(N, 10 * N.x)
         >>> p1.vel(N)
         10*N.x
+        >>> p = Point('p')
+        >>> q = dynamicsymbols('q')
+        >>> p.set_vel(N, 10 * N.x)
+        >>> p2 = Point('p2')
+        >>> p2.set_pos(p, q*N.x)
+        >>> p2.vel(N)
+        (Derivative(q(t), t) + 10)*N.x
 
         """
 
         _check_frame(frame)
         if not (frame in self._vel_dict):
-            raise ValueError('Velocity of point ' + self.name + ' has not been'
+            visited = []
+            queue = [self]
+            while queue: #BFS to find nearest point
+                node = queue.pop(0)
+                if node not in visited:
+                    visited.append(node)
+                    for neighbor, neighbor_pos in node._pos_dict.items():
+                        try:
+                            neighbor_pos.express(frame) #Checks if pos vector is valid
```

Comment:
You don't have a test that explicitly tests this line. None of your tests include velocity expressions that are written as components expressed in different frames. You could test this by having a position vector expressed in multiple frames, try to calc the vel, it fails, and then add the orientation info between the frames, so that it can be expressed.

Reply:
Actually, I now see that your very last test, does check this line explicitly.

Reply:
So do I need to add separate test?

Reply:
No, you have one.

Reply:
But that lacked the orient part so I added a new specific test

Reply:
Ok, more tests are good.